documentation/C++style.md

```diff
@@ -296,6 +297,23 @@ like `Http`.
 Consistency is one of many aspects in the pursuit of clarity,
 but not an end in itself.
 
+## <a name="checklist"></a>Pull request checklist
+Before submitting a pull request, review the following items:
+*  Verify that all files have a license.
+*  Make sure that all source lines have 80 or fewer characters.
+*  Run `git diff` on all files to look for spurious changes such as `include <iostream>`.
+*  Run clang-format version 7 on all .cc and .h files.
+*  Build and test with both GNU and clang compilers.
+*  Make sure that all error messages you added appear in tests.
+*  Annotate the code and tests with appropriate references to constraint and requirement numbers in the standard.
+*  Review declarations for proper use of `constexpr` and `const`.
+*  Ensure that the names of C++ constructs (functions, classes, etc.) evoke their purpose.
+*  Ensure that you check all dereferences of non-owning pointers.
+*  Ensure that the scopes of all functions and variables are as local as possible.
+*  Try to make all functions fit on a screen (40 lines).
+*  Declare non-member functions static when possible.
```

Comment:
https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

Reply:
The point I was trying to make about error messages is as follows.  When you add code to the compiler that produces an error message when processing Fortran source (as happens when you add a semantic check), you should add a test to the test suite that tests for the appearance of that message.

Thanks for the reference to using anonymous namespaces.  I didn't even realize that they existed.  But after reading that section of the LLVM coding standards, they don't seem like something that we should encourage.